src/ctia/domain/id.clj

```diff
@@ -70,13 +73,14 @@
 
 (defn short-id->id
   [type short-id {:keys [hostname path-prefix port protocol]}]
-  (map->CtiaId
-   {:hostname hostname
-    :short-id short-id
-    :path-prefix (not-empty path-prefix)
-    :port port
-    :protocol protocol
-    :type (name type)}))
+  (if (valid-short-id? short-id)
```

Comment:
I used to use `when` exclusively in spots like this. Then one time I contributed a PR to one of Technomancy's projects and he shared his opinion which was that you should avoid `when` because it's implicit `do` implies side effects. (and side effects are something to be avoided)

I personally, don't generally concern myself with it anymore :) Simply because there are valid arguments to both sides and it mostly is a very small matter.